src/MmTootleTool.cpp

```diff
@@ -82,36 +83,96 @@ namespace meshmagick
 	void FillMeshData(v1::HardwareIndexBufferSharedPtr indexBuffer,
 		v1::VertexDeclaration *vertexDeclaration,
 		v1::VertexBufferBinding* vertexBufferBinding,
-		std::vector<Vector3> & vertices,
-		std::vector<unsigned int> &indices)
+		std::vector<UniqueVertex> & vertices,
+		std::vector<unsigned int> &indices,
+		size_t numvertices)
 	{
-		// Get the target element
-		const v1::VertexElement* srcElem = vertexDeclaration->findElementBySemantic(
-			VES_POSITION);
-
-		if (!srcElem || srcElem->getType() != VET_FLOAT3)
+		// Lock all the buffers first
+		typedef std::vector<char*> BufferLocks;
+		BufferLocks bufferLocks;
+		const v1::VertexBufferBinding::VertexBufferBindingMap& bindings =
```

Comment:
please use `const auto&` here or put `using v1::VertexBufferBinding;` at the top. This will greatly help keeping master and this branch in sync.

Reply:
I'll do it tomorrow. Otherwise, if you prefer, the repository is public so you're able to modify the code by your own.
What do you prefer? `const auto` is not available in C++98 and OGRE 2.x still uses that version, right? But TBH, I've seen another PR which made use of `auto` ignoring this consideration.

Reply:
> C++98 and OGRE 2.x still uses that version, right?

yeah, but this choice does not affect external projects like this. So you are free to use c++11 here.

Reply:
Done: https://github.com/OGRECave/meshmagick/pull/4/commits/5eb4eb7f7a73780b2dfa5ce9bbbb51ccd3ea1bea